### PR TITLE
randomWRR: remove lock for accessing WRR.items 

### PR DESCRIPTION
**Pull request by arvindbr8**
`Add()` is always called as part of constructing the WRR object and never concurrently. Also turns out `Next()` is never called concurrently either. I started the PR (#5970) started out to avoid the race when accessing WRR.items in `String()`, but the mutex can be avoided all together for random WRR.

Also updating docstring to say the same thing.

RELEASE NOTES: N/A

### CODE REVIEW

File: internal/wrr/random_test.go
Comment (easwars) [inline internal/wrr/random_test.go:1]: Why is this file empty?

Comment (dfawley) [reply internal/wrr/random_test.go:1]: +1

File: internal/wrr/random.go [hunk 1]
```diff
@@ -87,5 +87,7 @@ func (rw *randomWRR) Add(item any, weight int64) {
 }
 
 func (rw *randomWRR) String() string {
+	rw.mu.Lock()
+	defer rw.mu.Unlock()
```
Comment (dfawley) [inline internal/wrr/random.go:91]: This is risky as we'll deadlock if `String` is ever called while the lock is held.

Note that `String` is called indirectly, too, e.g. `fmt.Sprint(rw)`.  I don't think I like this.  Maybe we should just exclude the `items` from what is printed and print only static info instead, with a different method that can retrieve the detailed data as a string if needed.

Consider:

```go
func (rw *randomWRR) Whatever() {
	rw.mu.Lock()
	defer rw.mu.Unlock()
	...
	if badThing {
		rw.logger.Warningf("saw bad thing in %s", rw)
	}
	// or:
	if rw.logger.V(2) {
		rw.logger.Infof("saw weird thing in %s", rw)
	}
}
```

We'd deadlock in rare cases that might not be covered in tests.

File: internal/wrr/random_test.go
Comment (arvindbr8) [reply internal/wrr/random_test.go:1]: oops. I had initially thought about adding a unit test for this package. But i realized this was basically covered by other tests indirectly. Removed

File: internal/wrr/random.go (hunk 1 above)
Comment (arvindbr8) [reply internal/wrr/random.go:91]: I just noticed that `rw.mu` is of type `sync.RWMutex`. Which means i could prolly use `mu.RLock()` in String(), which would avoid deadlocks on the mutex? wdyt? @dfawley

Comment (arvindbr8) [reply internal/wrr/random.go:91]: Spoke with him offline.. my b, rLock still blocks on wLock.

**Comment by arvindbr8** [comment]
Life happens, and pull request intents change.

Discussed with Doug offline. `Add()` is always called as part of constructing the WRR object and never concurrently. Also turns out `Next()` is never called concurrently either. I started the PR started out to avoid the race when accessing WRR.items in String() to fix https://github.com/grpc/grpc-go/issues/5970, the mutex can be avoided all together for random WRR. 

Updating the PR description

File: internal/wrr/wrr.go [hunk 2]
```diff
@@ -21,12 +21,11 @@ package wrr
 
 // WRR defines an interface that implements weighted round robin.
 type WRR interface {
-	// Add adds an item with weight to the WRR set.
-	//
-	// Add and Next need to be thread safe.
+	// Add adds an item with weight to the WRR set. All Add must be only called
+	// before any calls to Next().
```
Comment (dfawley) [inline internal/wrr/wrr.go:25]: Nits:

s/All//

s/Next()/Next/

File: internal/wrr/wrr.go [hunk 3]
```diff
@@ -21,12 +21,11 @@ package wrr
...
 	Add(item any, weight int64)
 	// Next returns the next picked item.
 	//
-	// Add and Next need to be thread safe.
+	// Next needs to be thread safe.
```
Comment (dfawley) [inline internal/wrr/wrr.go:29]: Possibly for clarity: "Add may not be called after any call to Next."

**Review by dfawley — APPROVED** [review]

File: internal/wrr/wrr.go (hunk 2 above)
Comment (arvindbr8) [reply internal/wrr/wrr.go]: Done.

File: internal/wrr/wrr.go (hunk 3 above)
Comment (arvindbr8) [reply internal/wrr/wrr.go]: Done.

**Pull request merged**